### PR TITLE
dev-container: add Dockerfile to build a dev container

### PR DESCRIPTION
**Pull request by jiangliu**
Add Dockerfile to build a container image for td-shim development
environment.

Signed-off-by: Liu Jiang <gerry@linux.alibaba.com>

### CODE REVIEW

File: devtools/dev_container/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+# https://docs.docker.com/develop/develop-images/dockerfile_best-practices/#run
+# DEBIAN_FRONTEND is set for tzdata.
+RUN apt-get update && \
+    DEBIAN_FRONTEND="noninteractive" apt-get install --no-install-recommends -y \
+    build-essential ca-certificates curl gcc git libssl-dev pkg-config ssh \
+    clang llvm nasm \
+    # cleanup
+    && apt-get clean && rm -rf /var/lib/apt/lists/*
+
+# Install rustup and a fixed version of Rust.
```
Comment (c3d) [inline devtools/dev_container/Dockerfile:18]: I don't mind, but shouldn't you do a run with multiple steps just to be consistent with the comments on lines 8-10? 😅

**Review by c3d — APPROVED** [review]
Thanks @jiangliu . It's so easy to review and approve such smaller commits 😀 I really appreciate that it's quite a bit of extra work on your side, but it makes it so much easier for us.

File: devtools/dev_container/Dockerfile (hunk 1 above)
Comment (jiangliu) [reply devtools/dev_container/Dockerfile:18]: The dev container is unstable yet, so keeping them in separate lines helps to use the build cache.
We may combine them into one line once it gets stable:)

**Comment by jiangliu** [comment]
> Thanks @jiangliu . It's so easy to review and approve such smaller commits 😀 I really appreciate that it's quite a bit of extra work on your side, but it makes it so much easier for us.

Never minder, it's also a great chance for me to get familiar with the code too:)

File: devtools/dev_container/Dockerfile (hunk 1 above)
Comment (c3d) [reply devtools/dev_container/Dockerfile:18]: OK. Makes sense.

**Review by jyao1 — APPROVED** [review]

**Pull request merged**